### DISCUSSION TITLE
[quectel0] Error reading SMS message

### DISCUSSION THREAD

**Discussion by grelasroberto** [IchthysMaranatha/asterisk-chan-quectel#26] [General]
Started stress testing sms receiving.. after some time,  Ive go this error:

[Mar  9 13:16:10] ERROR[31002]: at_response.c:426 log_cmd_response_error: [quectel0] Error reading SMS message
[Mar  8 11:34:37] DEBUG[15552] at_response.c: [quectel0] Error reading SMS message
[Mar  8 11:34:37] DEBUG[15552] at_queue.c: [quectel0] insert task with 1 commands begin with 'AT+CMGR' expected response '+CMGR' at tail of queue
[Mar  8 11:34:37] DEBUG[15552] at_queue.c: [quectel0] remove command 'AT+CMGR' expected response '+CMGR' real 'ERROR' cmd 1/1 flags 0x04 from queue
[Mar  8 11:34:37] DEBUG[15552] at_queue.c: [quectel0] remove task with 1 command(s) begin with 'AT+CMGR' expected response '+CMGR' from queue
[Mar  8 11:34:37] DEBUG[15552] at_queue.c: [quectel0] write command 'AT+CMGR' expected response '+CMGR' length 11
[Mar  8 11:34:37] DEBUG[15552] at_queue.c: [quectel0] [AT+CMGR=42^M]
[Mar  8 11:34:37] DEBUG[15552] at_read.c: [quectel0] receive 9 byte, used 9, free 2039, read 0, write 9
[Mar  8 11:34:37] DEBUG[15552] at_read.c: [quectel0] [^M
ERROR^M
]
[Mar  8 11:34:37] DEBUG[15552] at_response.c: [quectel0] Command response error suppressed:
[Mar  8 11:34:37] DEBUG[15552] at_response.c: [quectel0] Error reading SMS message
[Mar  8 11:34:37] DEBUG[15552] at_queue.c: [quectel0] insert task with 1 commands begin with 'AT+CMGR' expected response '+CMGR' at tail of queue
[Mar  8 11:34:37] DEBUG[15552] at_queue.c: [quectel0] remove command 'AT+CMGR' expected response '+CMGR' real 'ERROR' cmd 1/1 flags 0x04 from queue
[Mar  8 11:34:37] DEBUG[15552] at_queue.c: [quectel0] remove task with 1 command(s) begin with 'AT+CMGR' expected response '+CMGR' from queue
[Mar  8 11:34:37] DEBUG[15552] at_queue.c: [quectel0] write command 'AT+CMGR' expected response '+CMGR' length 11
[Mar  8 11:34:37] DEBUG[15552] at_queue.c: [quectel0] [AT+CMGR=43^M]

Then, it floods the log like sending  AT+CMGR=n   being n incremental by one till AT+CMGR=255, then looks like it reinitialize/resets the modem:

[Mar  8 11:34:45] DEBUG[15552] at_response.c: [quectel0] Provider query successfully
[Mar  8 11:34:45] DEBUG[15552] at_queue.c: [quectel0] remove command 'AT+COPS?' expected response 'OK' real 'OK' cmd 1/1 flags 0x01 from queue
[Mar  8 11:34:45] DEBUG[15552] at_queue.c: [quectel0] remove task with 1 command(s) begin with 'AT+COPS?' expected response 'OK' from queue
[Mar  8 11:34:55] DEBUG[15552] at_queue.c: [quectel0] insert task with 1 commands begin with 'AT' expected response 'OK' after head of queue
[Mar  8 11:34:55] DEBUG[15552] at_queue.c: [quectel0] write command 'AT' expected response 'OK' length 3
[Mar  8 11:34:55] DEBUG[15552] at_queue.c: [quectel0] [AT^M]
[Mar  8 11:34:55] DEBUG[15552] at_read.c: [quectel0] receive 6 byte, used 6, free 2042, read 0, write 6
[Mar  8 11:34:55] DEBUG[15552] at_read.c: [quectel0] [^M

Any clues ?  thank you!

**Comment by mpmc** [comment]
Same for me. I guess the format the device outputs isn't what is expected. Looking at the code now.
